src/base/bittorrent/session.cpp

```diff
@@ -4717,6 +4718,32 @@ void Session::startUpTorrents()
             }
         }
 
+        const TagSet tags = resumeData.tags;
```

Comment:
```suggestion
        const TagSet &tags = resumeData.tags;
```

Reply:
It is a bad idea to iterate over container using range based `for` and modify container inside loop body.

Reply:
Did you miss the comment above?

Reply:
>It is a bad idea to iterate over container using range based for and modify container inside loop body.

Then `Algorithm::removeIf()` can be of help here.
```c++
const TagSet &tags = resumeData.tags;
Algorithm::removeIf(resumeData.tags, [this](const QString &tag)
{
    if (hasTag(tag))
    	return false;

    if (addTag(tag))
    {
        LogMsg(tr("Detected inconsistent data: tag is missing from the configuration file."
                  " Tag will be recovered."
                  " Torrent: \"%1\". Tag: \"%2\"").arg(torrentID.toString(), tag), Log::WARNING);
        return false;
    }
    else
    {
        LogMsg(tr("Detected inconsistent data: invalid tag. Torrent: \"%1\". Tag: \"%2\"")
               .arg(torrentID.toString(), tag), Log::WARNING);
        return true;
    }
});
```

Reply:
`else` is redundant after `return`